app/knigths/knigth.py

```diff
@@ -0,0 +1,24 @@
+from __future__ import annotations
+from app.util.armour import get_armor
+from app.util.weapon import get_weapon
+
+
+class Knight:
+
+    def __init__(self, knight: dict) -> None:
+        self.name = knight["name"]
+        self.hp = knight["hp"]
+        self.power = get_weapon(knight)
+        self.protection = get_armor(knight)
+        self.get_potion(knight)
+
+    def get_potion(self, knight: dict) -> None:
+        if knight["potion"] is not None:
+            if "power" in knight["potion"]["effect"]:
+                self.power += knight["potion"]["effect"]["power"]
+
+            if "protection" in knight["potion"]["effect"]:
+                self.protection += knight["potion"]["effect"]["protection"]
+
+            if "hp" in knight["potion"]["effect"]:
+                self.hp += knight["potion"]["effect"]["hp"]
```

Comment:
Use loop instead of duplicate code

Reply:
How to write values in different attributes names?

Reply:
try something like this 

```
if knight["potion"]:
    potion_effect = knight["potion"]["effect"]
    for effect in potion_effect:
        if effect == "power":
            self.power += potion_effect[effect]
        elif effect == "protection":
            self.protection += potion_effect[effect]
        elif effect == "hp":
            self.hp += potion_effect[effect]
```

or this 
```
if knight["potion"]:
    potion_effect = knight["potion"]["effect"]
    self.power += potion_effect.get("power", 0) if "power" in potion_effect else 0
    self.protection += potion_effect.get("protection", 0) if "protection" in potion_effect else 0
    self.hp += potion_effect.get("hp", 0) if "hp" in potion_effect else 0
```